### PR TITLE
Bug/1.11 remove some warnings

### PR DESCRIPTION
**Pull request by xavierh**
Review please

### CODE REVIEW

File: src/main/java/com/minecolonies/coremod/placementhandlers/PlacementHandlers.java [hunk 1]
```diff
@@ -226,7 +227,7 @@ public Object handle(
     public static class FlowerPotPlacementHandler implements IPlacementHandler
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/coremod/placementhandlers/PlacementHandlers.java:226]: MAJOR Reduce this class from 31 to the maximum allowed 30 or externalize it in a public class. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2972)

File: src/main/java/com/minecolonies/coremod/placementhandlers/PlacementHandlers.java [hunk 2]
```diff
@@ -1,6 +1,7 @@
 package com.minecolonies.coremod.placementhandlers;
 
 import com.minecolonies.coremod.blocks.BlockSolidSubstitution;
+import com.minecolonies.coremod.colony.jobs.AbstractJob;
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/coremod/placementhandlers/PlacementHandlers.java]: MINOR Remove this unused import 'com.minecolonies.coremod.colony.jobs.AbstractJob'. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AUselessImportCheck)

File: src/main/java/com/minecolonies/coremod/placementhandlers/PlacementHandlers.java [hunk 3]
```diff
@@ -58,7 +59,7 @@ private PlacementHandlers()
     public static class FireplacementHandler implements IPlacementHandler
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/coremod/placementhandlers/PlacementHandlers.java:58]: MAJOR Reduce this class from 31 to the maximum allowed 30 or externalize it in a public class. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2972)

File: src/main/java/com/minecolonies/coremod/placementhandlers/IPlacementHandler.java [hunk 4]
```diff
@@ -1,5 +1,6 @@
 package com.minecolonies.coremod.placementhandlers;
 
+import com.minecolonies.coremod.colony.jobs.AbstractJob;
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/coremod/placementhandlers/IPlacementHandler.java]: MINOR Remove this unused import 'com.minecolonies.coremod.colony.jobs.AbstractJob'. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AUselessImportCheck)

File: src/main/java/com/minecolonies/coremod/items/ItemScepterPermission.java [hunk 5]
```diff
@@ -133,7 +133,7 @@ public EnumActionResult onItemUse(
 
         toggleItemMode(playerIn, compound);
 
-        return new ActionResult(EnumActionResult.SUCCESS, scepter);
+        return new ActionResult<ItemStack>(EnumActionResult.SUCCESS, scepter);
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/coremod/items/ItemScepterPermission.java]: MINOR Replace the type specification in this constructor call with the diamond operator ("<>"). (sonar.java.source not set. Assuming 7 or greater.) [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2293)

File: src/main/java/com/minecolonies/coremod/items/ItemScepterPermission.java [hunk 6]
```diff
@@ -122,7 +122,7 @@ public EnumActionResult onItemUse(
 
         if (worldIn.isRemote)
         {
-            return new ActionResult(EnumActionResult.SUCCESS, scepter);
+            return new ActionResult<ItemStack>(EnumActionResult.SUCCESS, scepter);
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/coremod/items/ItemScepterPermission.java]: MINOR Replace the type specification in this constructor call with the diamond operator ("<>"). (sonar.java.source not set. Assuming 7 or greater.) [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2293)

**Comment by MinecoloniesSonar** [comment]
SonarQube analysis reported 49 issues
* ![MAJOR][MAJOR] 3 major
* ![MINOR][MINOR] 46 minor

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [EntityAIStructureBuilder.java#L185](https://github.com/Minecolonies/minecolonies/blob/8f611674bf5b83e84017a4c71c66e90b2f3d6bb2/src/main/java/com/minecolonies/coremod/entity/ai/citizen/builder/EntityAIStructureBuilder.java#L185): The Cyclomatic Complexity of this method "requestMaterials" is 16 which is greater than 15 authorized. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)
1. ![MINOR][MINOR] [blockout](https://github.com/Minecolonies/minecolonies/blob/8f611674bf5b83e84017a4c71c66e90b2f3d6bb2/src/main/java/com/minecolonies/blockout): Add a 'package-info.java' file to document the 'blockout' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [controls](https://github.com/Minecolonies/minecolonies/blob/8f611674bf5b83e84017a4c71c66e90b2f3d6bb2/src/main/java/com/minecolonies/blockout/controls): Add a 'package-info.java' file to document the 'controls' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [views](https://github.com/Minecolonies/minecolonies/blob/8f611674bf5b83e84017a4c71c66e90b2f3d6bb2/src/main/java/com/minecolonies/blockout/views): Add a 'package-info.java' file to document the 'views' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [blocks](https://github.com/Minecolonies/minecolonies/blob/8f611674bf5b83e84017a4c71c66e90b2f3d6bb2/src/main/java/com/minecolonies/coremod/blocks): Add a 'package-info.java' file to document the 'blocks' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [gui](https://github.com/Minecolonies/minecolonies/blob/8f611674bf5b83e84017a4c71c66e90b2f3d6bb2/src/main/java/com/minecolonies/coremod/client/gui): Add a 'package-info.java' file to document the 'gui' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [render](https://github.com/Minecolonies/minecolonies/blob/8f611674bf5b83e84017a4c71c66e90b2f3d6bb2/src/main/java/com/minecolonies/coremod/client/render): Add a 'package-info.java' file to document the 'render' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [colony](https://github.com/Minecolonies/minecolonies/blob/8f611674bf5b83e84017a4c71c66e90b2f3d6bb2/src/main/java/com/minecolonies/coremod/colony): Add a 'package-info.java' file to document the 'colony' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [buildings](https://github.com/Minecolonies/minecolonies/blob/8f611674bf5b83e84017a4c71c66e90b2f3d6bb2/src/main/java/com/minecolonies/coremod/colony/buildings): Add a 'package-info.java' file to document the 'buildings' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [jobs](https://github.com/Minecolonies/minecolonies/blob/8f611674bf5b83e84017a4c71c66e90b2f3d6bb2/src/main/java/com/minecolonies/coremod/colony/jobs): Add a 'package-info.java' file to document the 'jobs' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by marvin-bitterlich** [comment]
@xavierh can we start with the `package-info.java` issues here?

**Comment by MinecoloniesSonar** [comment]
SonarQube analysis reported 68 issues
* ![MAJOR][MAJOR] 9 major
* ![MINOR][MINOR] 53 minor
* ![INFO][INFO] 6 info

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [EntityAIStructureBuilder.java#L185](https://github.com/Minecolonies/minecolonies/blob/7d0be8c80a81b33bbe128599d1ee598c70c232f5/src/main/java/com/minecolonies/coremod/entity/ai/citizen/builder/EntityAIStructureBuilder.java#L185): The Cyclomatic Complexity of this method "requestMaterials" is 16 which is greater than 15 authorized. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)
1. ![MAJOR][MAJOR] [InventoryUtils.java](https://github.com/Minecolonies/minecolonies/blob/7d0be8c80a81b33bbe128599d1ee598c70c232f5/src/main/java/com/minecolonies/coremod/util/InventoryUtils.java): This file has 780 lines, which is greater than 750 authorized. Split it into smaller files. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS00104)
1. ![MAJOR][MAJOR] [InventoryUtils.java#L26](https://github.com/Minecolonies/minecolonies/blob/7d0be8c80a81b33bbe128599d1ee598c70c232f5/src/main/java/com/minecolonies/coremod/util/InventoryUtils.java#L26): class "InventoryUtils" has 85 methods, which is greater than the 35 authorized. Split it into smaller classes. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1448)
1. ![MAJOR][MAJOR] [InventoryUtils.java#L1511](https://github.com/Minecolonies/minecolonies/blob/7d0be8c80a81b33bbe128599d1ee598c70c232f5/src/main/java/com/minecolonies/coremod/util/InventoryUtils.java#L1511): Merge this if statement with the enclosing one. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1066)
1. ![MAJOR][MAJOR] [InventoryUtils.java#L1514](https://github.com/Minecolonies/minecolonies/blob/7d0be8c80a81b33bbe128599d1ee598c70c232f5/src/main/java/com/minecolonies/coremod/util/InventoryUtils.java#L1514): Merge this if statement with the enclosing one. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1066)
1. ![MAJOR][MAJOR] [InventoryUtils.java#L1627](https://github.com/Minecolonies/minecolonies/blob/7d0be8c80a81b33bbe128599d1ee598c70c232f5/src/main/java/com/minecolonies/coremod/util/InventoryUtils.java#L1627): Assign this magic number 3 to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
1. ![MAJOR][MAJOR] [InventoryUtils.java#L1644](https://github.com/Minecolonies/minecolonies/blob/7d0be8c80a81b33bbe128599d1ee598c70c232f5/src/main/java/com/minecolonies/coremod/util/InventoryUtils.java#L1644): Remove this unused private "storePartialItemStack" method. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AUnusedPrivateMethod)
1. ![MINOR][MINOR] [blockout](https://github.com/Minecolonies/minecolonies/blob/7d0be8c80a81b33bbe128599d1ee598c70c232f5/src/main/java/com/minecolonies/blockout): Add a 'package-info.java' file to document the 'blockout' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [controls](https://github.com/Minecolonies/minecolonies/blob/7d0be8c80a81b33bbe128599d1ee598c70c232f5/src/main/java/com/minecolonies/blockout/controls): Add a 'package-info.java' file to document the 'controls' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [views](https://github.com/Minecolonies/minecolonies/blob/7d0be8c80a81b33bbe128599d1ee598c70c232f5/src/main/java/com/minecolonies/blockout/views): Add a 'package-info.java' file to document the 'views' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by xavierh** [comment]
@Kostronor sure, I will add the package info for the package which I modified, but only if I am confident enough to add a short description

**Comment by MinecoloniesSonar** [comment]
SonarQube analysis reported 65 issues
* ![MAJOR][MAJOR] 6 major
* ![MINOR][MINOR] 53 minor
* ![INFO][INFO] 6 info

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [EntityAIStructureBuilder.java#L185](https://github.com/Minecolonies/minecolonies/blob/54ac8468e5d54171b9147e3acc962dae4f6dcaee/src/main/java/com/minecolonies/coremod/entity/ai/citizen/builder/EntityAIStructureBuilder.java#L185): The Cyclomatic Complexity of this method "requestMaterials" is 16 which is greater than 15 authorized. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)
1. ![MAJOR][MAJOR] [InventoryUtils.java#L26](https://github.com/Minecolonies/minecolonies/blob/54ac8468e5d54171b9147e3acc962dae4f6dcaee/src/main/java/com/minecolonies/coremod/util/InventoryUtils.java#L26): class "InventoryUtils" has 85 methods, which is greater than the 35 authorized. Split it into smaller classes. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1448)
1. ![MAJOR][MAJOR] [InventoryUtils.java#L1617](https://github.com/Minecolonies/minecolonies/blob/54ac8468e5d54171b9147e3acc962dae4f6dcaee/src/main/java/com/minecolonies/coremod/util/InventoryUtils.java#L1617): Assign this magic number 3 to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
1. ![MAJOR][MAJOR] [InventoryUtils.java#L1634](https://github.com/Minecolonies/minecolonies/blob/54ac8468e5d54171b9147e3acc962dae4f6dcaee/src/main/java/com/minecolonies/coremod/util/InventoryUtils.java#L1634): Remove this unused private "storePartialItemStack" method. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AUnusedPrivateMethod)
1. ![MINOR][MINOR] [blockout](https://github.com/Minecolonies/minecolonies/blob/54ac8468e5d54171b9147e3acc962dae4f6dcaee/src/main/java/com/minecolonies/blockout): Add a 'package-info.java' file to document the 'blockout' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [controls](https://github.com/Minecolonies/minecolonies/blob/54ac8468e5d54171b9147e3acc962dae4f6dcaee/src/main/java/com/minecolonies/blockout/controls): Add a 'package-info.java' file to document the 'controls' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [views](https://github.com/Minecolonies/minecolonies/blob/54ac8468e5d54171b9147e3acc962dae4f6dcaee/src/main/java/com/minecolonies/blockout/views): Add a 'package-info.java' file to document the 'views' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [blocks](https://github.com/Minecolonies/minecolonies/blob/54ac8468e5d54171b9147e3acc962dae4f6dcaee/src/main/java/com/minecolonies/coremod/blocks): Add a 'package-info.java' file to document the 'blocks' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [gui](https://github.com/Minecolonies/minecolonies/blob/54ac8468e5d54171b9147e3acc962dae4f6dcaee/src/main/java/com/minecolonies/coremod/client/gui): Add a 'package-info.java' file to document the 'gui' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [render](https://github.com/Minecolonies/minecolonies/blob/54ac8468e5d54171b9147e3acc962dae4f6dcaee/src/main/java/com/minecolonies/coremod/client/render): Add a 'package-info.java' file to document the 'render' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by MinecoloniesSonar** [comment]
SonarQube analysis reported 62 issues
* ![MAJOR][MAJOR] 5 major
* ![MINOR][MINOR] 51 minor
* ![INFO][INFO] 6 info

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [EntityAIStructureBuilder.java#L185](https://github.com/Minecolonies/minecolonies/blob/f3a749284fff89279a764ac4d404a4ebd8abf3e3/src/main/java/com/minecolonies/coremod/entity/ai/citizen/builder/EntityAIStructureBuilder.java#L185): The Cyclomatic Complexity of this method "requestMaterials" is 16 which is greater than 15 authorized. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)
1. ![MAJOR][MAJOR] [InventoryUtils.java#L26](https://github.com/Minecolonies/minecolonies/blob/f3a749284fff89279a764ac4d404a4ebd8abf3e3/src/main/java/com/minecolonies/coremod/util/InventoryUtils.java#L26): class "InventoryUtils" has 84 methods, which is greater than the 35 authorized. Split it into smaller classes. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1448)
1. ![MAJOR][MAJOR] [InventoryUtils.java#L1619](https://github.com/Minecolonies/minecolonies/blob/f3a749284fff89279a764ac4d404a4ebd8abf3e3/src/main/java/com/minecolonies/coremod/util/InventoryUtils.java#L1619): Assign this magic number 3 to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
1. ![MINOR][MINOR] [blockout](https://github.com/Minecolonies/minecolonies/blob/f3a749284fff89279a764ac4d404a4ebd8abf3e3/src/main/java/com/minecolonies/blockout): Add a 'package-info.java' file to document the 'blockout' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [controls](https://github.com/Minecolonies/minecolonies/blob/f3a749284fff89279a764ac4d404a4ebd8abf3e3/src/main/java/com/minecolonies/blockout/controls): Add a 'package-info.java' file to document the 'controls' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [views](https://github.com/Minecolonies/minecolonies/blob/f3a749284fff89279a764ac4d404a4ebd8abf3e3/src/main/java/com/minecolonies/blockout/views): Add a 'package-info.java' file to document the 'views' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [blocks](https://github.com/Minecolonies/minecolonies/blob/f3a749284fff89279a764ac4d404a4ebd8abf3e3/src/main/java/com/minecolonies/coremod/blocks): Add a 'package-info.java' file to document the 'blocks' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [gui](https://github.com/Minecolonies/minecolonies/blob/f3a749284fff89279a764ac4d404a4ebd8abf3e3/src/main/java/com/minecolonies/coremod/client/gui): Add a 'package-info.java' file to document the 'gui' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [render](https://github.com/Minecolonies/minecolonies/blob/f3a749284fff89279a764ac4d404a4ebd8abf3e3/src/main/java/com/minecolonies/coremod/client/render): Add a 'package-info.java' file to document the 'render' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [colony](https://github.com/Minecolonies/minecolonies/blob/f3a749284fff89279a764ac4d404a4ebd8abf3e3/src/main/java/com/minecolonies/coremod/colony): Add a 'package-info.java' file to document the 'colony' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

File: src/main/java/com/minecolonies/coremod/util/InventoryUtils.java [hunk 7]
```diff
@@ -1505,8 +1505,10 @@ public static boolean transferItemStackIntoNextFreeSlotInItemHandlers(@NotNull f
...
     {
-        if (!isItemStackEmpty(itemStack1) && !isItemStackEmpty(itemStack2) &&
-           itemStack1.getItem() == itemStack2.getItem() && itemStack1.getItemDamage() == itemStack2.getItemDamage())
+        if (!isItemStackEmpty(itemStack1) &&
+            !isItemStackEmpty(itemStack2) &&
+            itemStack1.getItem() == itemStack2.getItem() &&
+            itemStack1.getItemDamage() == itemStack2.getItemDamage())
         {
             // Then sort on NBT
             if (itemStack1.hasTagCompound() && itemStack2.hasTagCompound())
```
Comment (marvin-bitterlich) [inline src/main/java/com/minecolonies/coremod/util/InventoryUtils.java]: also merge this please :)

**Comment by MinecoloniesSonar** [comment]
SonarQube analysis reported 66 issues
* ![MAJOR][MAJOR] 5 major
* ![MINOR][MINOR] 55 minor
* ![INFO][INFO] 6 info

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [EntityAIStructureBuilder.java#L185](https://github.com/Minecolonies/minecolonies/blob/7f4b8df27113186cc2feb9e626fc2b2d5d8720fd/src/main/java/com/minecolonies/coremod/entity/ai/citizen/builder/EntityAIStructureBuilder.java#L185): The Cyclomatic Complexity of this method "requestMaterials" is 16 which is greater than 15 authorized. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)
1. ![MAJOR][MAJOR] [InventoryUtils.java#L26](https://github.com/Minecolonies/minecolonies/blob/7f4b8df27113186cc2feb9e626fc2b2d5d8720fd/src/main/java/com/minecolonies/coremod/util/InventoryUtils.java#L26): class "InventoryUtils" has 84 methods, which is greater than the 35 authorized. Split it into smaller classes. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1448)
1. ![MAJOR][MAJOR] [InventoryUtils.java#L1619](https://github.com/Minecolonies/minecolonies/blob/7f4b8df27113186cc2feb9e626fc2b2d5d8720fd/src/main/java/com/minecolonies/coremod/util/InventoryUtils.java#L1619): Assign this magic number 3 to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
1. ![MINOR][MINOR] [blockout](https://github.com/Minecolonies/minecolonies/blob/7f4b8df27113186cc2feb9e626fc2b2d5d8720fd/src/main/java/com/minecolonies/blockout): Add a 'package-info.java' file to document the 'blockout' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [controls](https://github.com/Minecolonies/minecolonies/blob/7f4b8df27113186cc2feb9e626fc2b2d5d8720fd/src/main/java/com/minecolonies/blockout/controls): Add a 'package-info.java' file to document the 'controls' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [views](https://github.com/Minecolonies/minecolonies/blob/7f4b8df27113186cc2feb9e626fc2b2d5d8720fd/src/main/java/com/minecolonies/blockout/views): Add a 'package-info.java' file to document the 'views' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [blocks](https://github.com/Minecolonies/minecolonies/blob/7f4b8df27113186cc2feb9e626fc2b2d5d8720fd/src/main/java/com/minecolonies/coremod/blocks): Add a 'package-info.java' file to document the 'blocks' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [gui](https://github.com/Minecolonies/minecolonies/blob/7f4b8df27113186cc2feb9e626fc2b2d5d8720fd/src/main/java/com/minecolonies/coremod/client/gui): Add a 'package-info.java' file to document the 'gui' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [render](https://github.com/Minecolonies/minecolonies/blob/7f4b8df27113186cc2feb9e626fc2b2d5d8720fd/src/main/java/com/minecolonies/coremod/client/render): Add a 'package-info.java' file to document the 'render' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [colony](https://github.com/Minecolonies/minecolonies/blob/7f4b8df27113186cc2feb9e626fc2b2d5d8720fd/src/main/java/com/minecolonies/coremod/colony): Add a 'package-info.java' file to document the 'colony' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

File: src/main/java/com/minecolonies/coremod/entity/ai/citizen/package-info.java [hunk 8]
```diff
@@ -0,0 +1,4 @@
+/**
+ * Contains the IA for the citizens.
```
Comment (marvin-bitterlich) [inline src/main/java/com/minecolonies/coremod/entity/ai/citizen/package-info.java]: AI ;)

Comment (xavierh) [reply src/main/java/com/minecolonies/coremod/entity/ai/citizen/package-info.java]: My french is kicking in, will fix that :)

**Comment by MinecoloniesSonar** [comment]
SonarQube analysis reported 68 issues
* ![MAJOR][MAJOR] 5 major
* ![MINOR][MINOR] 57 minor
* ![INFO][INFO] 6 info

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [EntityAIStructureBuilder.java#L185](https://github.com/Minecolonies/minecolonies/blob/7915b19ad1deebd8fc42207356c88a3aa137dea6/src/main/java/com/minecolonies/coremod/entity/ai/citizen/builder/EntityAIStructureBuilder.java#L185): The Cyclomatic Complexity of this method "requestMaterials" is 16 which is greater than 15 authorized. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)
1. ![MAJOR][MAJOR] [InventoryUtils.java#L26](https://github.com/Minecolonies/minecolonies/blob/7915b19ad1deebd8fc42207356c88a3aa137dea6/src/main/java/com/minecolonies/coremod/util/InventoryUtils.java#L26): class "InventoryUtils" has 84 methods, which is greater than the 35 authorized. Split it into smaller classes. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1448)
1. ![MAJOR][MAJOR] [InventoryUtils.java#L1619](https://github.com/Minecolonies/minecolonies/blob/7915b19ad1deebd8fc42207356c88a3aa137dea6/src/main/java/com/minecolonies/coremod/util/InventoryUtils.java#L1619): Assign this magic number 3 to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
1. ![MINOR][MINOR] [blockout](https://github.com/Minecolonies/minecolonies/blob/7915b19ad1deebd8fc42207356c88a3aa137dea6/src/main/java/com/minecolonies/blockout): Add a 'package-info.java' file to document the 'blockout' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [controls](https://github.com/Minecolonies/minecolonies/blob/7915b19ad1deebd8fc42207356c88a3aa137dea6/src/main/java/com/minecolonies/blockout/controls): Add a 'package-info.java' file to document the 'controls' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [views](https://github.com/Minecolonies/minecolonies/blob/7915b19ad1deebd8fc42207356c88a3aa137dea6/src/main/java/com/minecolonies/blockout/views): Add a 'package-info.java' file to document the 'views' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [blocks](https://github.com/Minecolonies/minecolonies/blob/7915b19ad1deebd8fc42207356c88a3aa137dea6/src/main/java/com/minecolonies/coremod/blocks): Add a 'package-info.java' file to document the 'blocks' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [gui](https://github.com/Minecolonies/minecolonies/blob/7915b19ad1deebd8fc42207356c88a3aa137dea6/src/main/java/com/minecolonies/coremod/client/gui): Add a 'package-info.java' file to document the 'gui' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [render](https://github.com/Minecolonies/minecolonies/blob/7915b19ad1deebd8fc42207356c88a3aa137dea6/src/main/java/com/minecolonies/coremod/client/render): Add a 'package-info.java' file to document the 'render' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [colony](https://github.com/Minecolonies/minecolonies/blob/7915b19ad1deebd8fc42207356c88a3aa137dea6/src/main/java/com/minecolonies/coremod/colony): Add a 'package-info.java' file to document the 'colony' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by xavierh** [comment]
@Kostronor done

**Comment by marvin-bitterlich** [comment]
:shipit:

[Approved with PullApprove](https://pullapprove.com/Minecolonies/minecolonies/pull-request/965/?utm_source=github-pr&utm_medium=comment-badge&utm_campaign=Minecolonies/minecolonies)

**Comment by marvin-bitterlich** [comment]
@Raycoms

**Comment by Raycoms** [comment]
:shipit: 

[Approved with PullApprove](https://pullapprove.com/Minecolonies/minecolonies/pull-request/965/?utm_source=github-pr&utm_medium=comment-badge&utm_campaign=Minecolonies/minecolonies)

**Pull request merged**